app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/details/DetailViewExampleWithVideoBackgroundFragment.java

```diff
@@ -0,0 +1,221 @@
+            imageResId = extras.getInt(EXTRA_CARD, imageResId);
+        }
+        detailsOverview.setImageDrawable(getResources().getDrawable(imageResId, null));
+        ArrayObjectAdapter actionAdapter = new ArrayObjectAdapter();
+
+        mActionPlay = new Action(ACTION_PLAY, getString(R.string.action_play));
+        mActionBuy = new Action(ACTION_BUY, getString(R.string.action_buy) + data.getPrice());
+        mActionWishList = new Action(ACTION_WISHLIST, getString(R.string.action_wishlist));
+        mActionRelated = new Action(ACTION_RELATED, getString(R.string.action_related));
+
+        actionAdapter.add(mActionBuy);
+        actionAdapter.add(mActionWishList);
+        actionAdapter.add(mActionRelated);
+        detailsOverview.setActionsAdapter(actionAdapter);
+        mRowsAdapter.add(detailsOverview);
+
+        // Setup related row.
+        ArrayObjectAdapter listRowAdapter = new ArrayObjectAdapter(
+                new CardPresenterSelector(getActivity()));
+        for (Card characterCard : data.getCharacters()) listRowAdapter.add(characterCard);
+        HeaderItem header = new HeaderItem(0, getString(R.string.header_related));
+        mRowsAdapter.add(new CardListRow(header, listRowAdapter, null));
+
+        // Setup recommended row.
+        listRowAdapter = new ArrayObjectAdapter(new CardPresenterSelector(getActivity()));
+        for (Card card : data.getRecommended()) listRowAdapter.add(card);
+        header = new HeaderItem(1, getString(R.string.header_recommended));
+        mRowsAdapter.add(new ListRow(header, listRowAdapter));
+
+        setAdapter(mRowsAdapter);
+        new Handler().postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                startEntranceTransition();
+            }
+        }, 500);
+        updateBackground(data);
+    }
+
+    private void updateBackground(DetailedCard data) {
+        mDetailsBackground.enableParallax();
+
+        mMediaPlayerGlue = new MediaPlayerGlue(getActivity());
+        mDetailsBackground.setupVideoPlayback(mMediaPlayerGlue);
+
+        mMediaPlayerGlue.setTitle(data.getTitle().concat(" (Trailer)"));
+        mMediaPlayerGlue.setArtist(data.getDescription());
+        mMediaPlayerGlue.setVideoUrl(data.getTrailerUrl());
+    }
+
+    private void setupEventListeners() {
+        setOnItemViewSelectedListener(this);
+        setOnItemViewClickedListener(this);
+    }
+
+    @Override
+    public void onItemClicked(Presenter.ViewHolder itemViewHolder, Object item,
+                              RowPresenter.ViewHolder rowViewHolder, Row row) {
+        if (!(item instanceof Action)) return;
+        Action action = (Action) item;
+        long id = action.getId();
+
+        ArrayObjectAdapter actionAdapter = (ArrayObjectAdapter)
+                ((DetailsOverviewRow) getAdapter().get(0)).getActionsAdapter();
+
+        if (id == ACTION_PLAY) {
+            // TODO: play main video on background
```

Comment:
Since this sample app is in not in "package android.support.v17.leanback.app;", this class doesn't have permission to access no modifier variable mVideoFragment in DetailsFragment.